pocs/utils/messaging.py

```diff
@@ -217,9 +222,15 @@ def receive_message(self, blocking=True, flags=0):
         msg_obj = None
         if not blocking:
             flags = flags | zmq.NOBLOCK
+        elif timeout_ms > 0:
+            # Wait until a message is available or the timeout expires.
+            self.socket.poll(timeout=timeout_ms, flags=(zmq.POLLIN | zmq.POLLOUT))
+            # Don't block at this point, because we will have waited as long
+            # as necessary.
+            flags = flags | zmq.NOBLOCK
         try:
             message = self.socket.recv_string(flags=flags)
-        except Exception:
+        except Exception as e:
```

Comment:
I'm surprised pyflakes doesn't complain about not using the `e`. Any reason it was added?

Reply:
Good question. Probably because I was using the debugger. I'll double check pyflakes on this.